File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long>q = new PriorityQueue<>();
        for(long i : nums) q.add(i);
        int cnt = 0 ;
        while(q.size() > 1){
            long a = q.poll();
            long b = q.poll();
            long mn = Math.min(a, b);
            long mx = Math.max(a, b);
            if(mn >= k) break ;
            cnt ++ ;
            q.add(2*mn + mx);
        }
        return cnt ;
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations(self, q: List[int], k: int) -> int:
        heapq.heapify(q)
        
        ans = 0
        while len(q) > 1:
            a, b = heapq.heappop(q), heapq.heappop(q)
            mn, mx = min(a, b), max(a, b)
            if mn >= k: break 
            ans += 1
            heapq.heappush(q, 2*mn + mx)

        return ans 